File: repos_estudiantes_TARQUI_primera_entrega/RUBIN_DE_CELIS_VALDEZ_ANDRES/INTRODUCCION-POO/EJERCICIO-5/Estudiante.java
public class Estudiante 
{
    private String nombre;
    private double nota_1;
    private double nota_2;
    
    public Estudiante(String nombre, double nota_1, double nota_2)
    {
        this.nombre = nombre;
        this.nota_1 = nota_1;
        this.nota_2 = nota_2;
    }
    
    public double calcularPromedio() 
    {
        return (nota_1 + nota_2) / 2;
    }
    
    public boolean aprobo() 
    {
        return calcularPromedio() >= 6;
    }
  
    public static void main(String[] args) 
    {
        Estudiante est1 = new Estudiante("Ana", 7.5, 8.0);
        Estudiante est2 = new Estudiante("Luis", 5.0, 6.5);
        Estudiante est3 = new Estudiante("Sofia", 9.0, 9.5);

        System.out.println("Estudiante: " + est1.nombre);
        System.out.println("Promedio: " + est1.calcularPromedio());
        System.out.println("Aprobado: " + (est1.aprobo() ? "Si" : "No"));
        System.out.println();

        System.out.println("Estudiante: " + est2.nombre);
        System.out.println("Promedio: " + est2.calcularPromedio());
        System.out.println("Aprobado: " + (est2.aprobo() ? "Si" : "No"));
        System.out.println();

        System.out.println("Estudiante: " + est3.nombre);
        System.out.println("Promedio: " + est3.calcularPromedio());
        System.out.println("Aprobado: " + (est3.aprobo() ? "Si" : "No"));
    } 
}

File: repos_estudiantes_ALVAREZ_primera_entrega/CALLE_CASTILLO_CRISTHIAN_JAVIER/PRACTICA1-PROGRAMACION-II/PRACTICA INF-121/PROGRAMACION ORIENTADA A OBJETOS/EJERCICIO 1/Estudiante.py
class Estudiante:
    def __init__(self, nombre, nota_1, nota_2):
        self.nombre = nombre
        self.nota_1 = nota_1
        self.nota_2 = nota_2

    def calcular_promedio(self):
        return (self.nota_1 + self.nota_2) / 2

    def verificar_aprobado(self):
        return self.calcular_promedio() >= 6

    def mostrar_info(self):
        promedio = self.calcular_promedio()
        print(f"{self.nombre} tiene un promedio de {promedio:.2f}")
        if self.verificar_aprobado():
            print("Estado: Aprobado")
        else:
            print("Estado: Reprobado")
        print("-" * 30)

est1 = Estudiante("Juan", 7.5, 6.0)
est2 = Estudiante("Carla", 5.0, 4.5)
est3 = Estudiante("Luis", 8.0, 9.0)

for estudiante in [est1, est2, est3]:
    estudiante.mostrar_info()
